src/components/calcite-pick-list/calcite-pick-list.tsx

```diff
@@ -63,6 +63,8 @@ export class CalcitePickList {
 
   @State() selectedValues: Map<string, HTMLCalcitePickListItemElement> = new Map();
 
+  @State() dataForFilter: object[] = [];
```

Comment:
If we expect the filter to be a reusable component, then any component that uses the filter will have to do this `getItemData` method and `dataForFilter` state. 

It might make it much easier on the developer if this was handled by the Filter component itself.

For example:

```
CalciteFilter {
  referenceElement: HTMLElement; // The element used to query for filter items
  dataAttr: string; //  The attribute used to store the metadata. EG: 'metadata' 
  selector: string; // The selector to query for filter items. EG: "calcite-picker-item"
  matchedClass: string; // The class applied to matches. EG: "calcite-filter-matched"
  unmatchedClass: string; //  The class applied to non matches. EG: "calcite-filter-unmatched"
}
```

Reply:
@driskull Can we wait for more use cases to arise before extracting components?

Reply:
Yep!

Reply:
Yea there may be an opportunity here to tuck away more inside the filter internals, but we'll want to make sure we don't limit it's uses in doing so.